nd4j/nd4j-backends/nd4j-api-parent/nd4j-native-api/src/main/java/org/nd4j/nativeblas/NativeOpsGPUInfoProvider.java

```
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2018 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.nativeblas;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.nd4j.systeminfo.GPUInfo;
+import org.nd4j.systeminfo.GPUInfoProvider;
+
+public class NativeOpsGPUInfoProvider implements GPUInfoProvider {
+
+    @Override
+    public List<GPUInfo> getGPUs() {
+        NativeOps nativeOps = NativeOpsHolder.getInstance().getDeviceNativeOps();
+
+        List<GPUInfo> gpus = new ArrayList<>();
+
+
+        int nDevices = nativeOps.getAvailableDevices();
+        if (nDevices > 0) {
+            for (int i = 0; i < nDevices; i++) {
+                try {
+                    String name = nativeOps.getDeviceName(i);
+                    long total = nativeOps.getDeviceTotalMemory(i);
+                    long free = nativeOps.getDeviceFreeMemory(i);
+                    int major = nativeOps.getDeviceMajor(i);
+                    int minor = nativeOps.getDeviceMinor(i);
+
+                    gpus.add(new GPUInfo(name, total, free, major, minor));
+                } catch (Exception e) {
+                }
```

Maybe log.warn on exception - better the user knows something went wrong than silently failing...